nixos/modules/services/networking/pangolin/default.nix

```
@@ -0,0 +1,1270 @@
+                    middlewares = [ "redirect-to-https" ];
+                  };
+                  # Next.js router (handles everything except API and WebSocket paths)
+                  next-router = {
+                    rule = "Host(`${dashboardDomain}`) && !PathPrefix(`/api/v1`)";
+                    service = "next-service";
+                    entryPoints = [ pangolin.settings.traefik.https_entrypoint ];
+                    tls =
+                      optionalAttrs prefer_wildcard_cert {
+                        domains = [
+                          { main = base_domain; }
+                          { sans = "*.${base_domain}"; }
+                        ];
+                      }
+                      // {
+                        certResolver = cert_resolver;
+                      };
+                  };
+                  # API router (handles /api/v1 paths)
+                  api-router = {
+                    rule = "Host(`${dashboardDomain}`) && PathPrefix(`/api/v1`)";
+                    service = "api-service";
+                    entryPoints = [ pangolin.settings.traefik.https_entrypoint ];
+                    tls.certResolver = cert_resolver;
+                  };
+                  # WebSocket router
+                  ws-router = {
+                    rule = "Host(`${dashboardDomain}`)";
+                    service = "api-service";
+                    entryPoints = [ pangolin.settings.traefik.https_entrypoint ];
+                    tls.certResolver = cert_resolver;
+                  };
+                };
+                services = {
+                  # Next.JS server (dashboard)
+                  next-service.loadBalancer.servers = [
+                    {
+                      url = "http://localhost:${toString pangolin.settings.server.next_port}";
+                    }
+                  ];
+                  # API/WebSocket server
+                  api-service.loadBalancer.servers = [
+                    {
+                      url = "http://localhost:${toString pangolin.settings.server.external_port}";
+                    }
+                  ];
+                };
+              };
+            };
+          };
+      })
+      (mkIf gerbil.enable {
+        assertions = [
+          {
+            assertion = badger.enable;
+            message = ''
+              Badger must be enabled if Gerbil is enabled.
+              Please set 'services.badger.enable' to 'true'.
+            '';
+          }
+          {
+            assertion = !isNull gerbil.settings.reachableAt;
+            message = ''
+              'services.gerbil.settings.reachableAt' must be provided when Gerbil
+              is enabled.
+            '';
+          }
+          {
+            assertion =
+              (!isNull gerbil.settings.remoteConfig && isNull gerbil.settings.config)
+              || (isNull gerbil.settings.remoteConfig && !isNull gerbil.settings.config);
```

```suggestion
             ( gerbil.settings.remoteConfig ?  gerbil.settings.config ? false )
              && (isNull gerbil.settings.remoteConfig || isNull gerbil.settings.config);
```
gah, idk, but i feel like this can be sexier

---

This isn't valid syntax, and why would we do that? Both `config` and `remoteConfig` are defined in the submodule.

---

Now implemented via `xor`.